File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> hash = new HashMap<>();

        // 개수 파악
        for (String people : participant){
            if (hash.containsKey(people)) {
                Integer val = hash.get(people);
                hash.put(people, val+1);
            } else {
                hash.put(people, 1);
            }
        }
        // 완주한 사람의 개수를 빼줌
        for (String people : completion) {
            if (hash.containsKey(people)) {
                Integer val = hash.get(people);
                hash.put(people, val - 1);
            } else {
                return people;
            }
        }
        // 값이 1인것을 찾아내서 return
        for (String people : hash.keySet()) {
            if (hash.get(people) == 1) {
                return people;
            }
        }
        return "";
    }
}

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dic = {}
    for people in participant:
        if people in dic:
            dic[people] +=1
        else:
            dic[people] = 1
    
    for people in completion:
        if people in dic:
            dic[people] -=1
        else:
            return people  
    
    for people in dic.keys():
        if dic[people]:
            return people
    